plugin/evm/uptime/pausable_manager_test.go

```diff
@@ -0,0 +1,234 @@
+
+				// Disconnect and check uptime
+				currentTime = addTime(clk, 3*time.Second)
+				require.NoError(up.Disconnect(nodeID0))
+				// Uptime should not have increased since the node was paused
+				checkUptime(t, up, nodeID0, expectedUptime, currentTime)
+
+				// Connect again and resume
+				addTime(clk, 4*time.Second)
+				require.NoError(up.Connect(nodeID0))
+				addTime(clk, 5*time.Second)
+				up.OnValidatorStatusUpdated(vID, nodeID0, true)
+				require.False(up.IsPaused(nodeID0))
+
+				// Check uptime after resume
+				currentTime = addTime(clk, 6*time.Second)
+				// Uptime should have increased by 6 seconds since the node was resumed
+				expectedUptime += 6 * time.Second
+				checkUptime(t, up, nodeID0, expectedUptime, currentTime)
+			},
+		},
+		{
+			name: "Case 4: Start tracking, connect, pause, stop tracking, resume tracking",
+			testFunc: func(t *testing.T, up PausableManager, clk *mockable.Clock, s uptime.State) {
+				// Start tracking and connect
+				require.NoError(up.StartTracking([]ids.NodeID{nodeID0}))
+				addTime(clk, time.Second)
+				require.NoError(up.Connect(nodeID0))
+
+				// Pause and check uptime
+				currentTime := addTime(clk, 2*time.Second)
+				up.OnValidatorStatusUpdated(vID, nodeID0, false)
+				require.True(up.IsPaused(nodeID0))
+				// Uptime should be 2 seconds since the node was paused after 2 seconds
+				expectedUptime := 2 * time.Second
+
+				checkUptime(t, up, nodeID0, expectedUptime, currentTime)
+
+				// Stop tracking and reinitialize manager
+				currentTime = addTime(clk, 3*time.Second)
+				require.NoError(up.StopTracking([]ids.NodeID{nodeID0}))
+				checkUptime(t, up, nodeID0, expectedUptime, currentTime)
+				up = NewPausableManager(uptime.NewManager(s, clk))
+
+				// Uptime should not have increased since the node was paused
+				// and we have not started tracking again
+				checkUptime(t, up, nodeID0, expectedUptime, currentTime)
+
+				// Pause and check uptime
+				up.OnValidatorStatusUpdated(vID, nodeID0, false)
+				require.True(up.IsPaused(nodeID0))
+				// Uptime should not have increased since the node was paused
+				checkUptime(t, up, nodeID0, expectedUptime, currentTime)
+
+				// Resume and check uptime
+				currentTime = addTime(clk, 5*time.Second)
+				up.OnValidatorStatusUpdated(vID, nodeID0, true)
+				require.False(up.IsPaused(nodeID0))
+				// Uptime should have increased by 5 seconds since the node was resumed
+				expectedUptime += 5 * time.Second
+				checkUptime(t, up, nodeID0, expectedUptime, currentTime)
+
+				// Start tracking and check elapsed time
+				currentTime = addTime(clk, 6*time.Second)
+				require.NoError(up.StartTracking([]ids.NodeID{nodeID0}))
+				// Uptime should have increased by 6 seconds since we started tracking
+				// and node was resumed (we assume the node was online until we started tracking)
+				expectedUptime += 6 * time.Second
+				checkUptime(t, up, nodeID0, expectedUptime, currentTime)
```

Comment:
We assume all nodes are online when we go offline (until we start tracking again). In this case we have first resumed the node, and then started tracking again. It indicates the paused node was resumed while we're offline. So we should increase the uptime of the node. 

In between this and 194 it should not increase the uptime since the nodeID0 is not connected to the manager yet.